Source/Scene/PointCloudPostProcessor.js

```diff
@@ -0,0 +1,1178 @@
+                return processor._ecTextures[1];
+            },
+            u_occlusionAngle : function() {
+                return processor.occlusionAngle;
+            },
+            u_dropoutFactor : function() {
+                return processor.dropoutFactor;
+            }
+        };
+
+        var pointOcclusionFS = addConstants(
+            (context.webgl2) ? PointOcclusionPassGL2 : PointOcclusionPassGL1,
+            'neighborhoodHalfWidth',
+            processor.neighborhoodHalfWidth
+        );
+
+        pointOcclusionFS = addConstants(
+            pointOcclusionFS,
+            'C0',
+            processor.C0
+        );
+
+        pointOcclusionFS = addConstants(
+            pointOcclusionFS,
+            'C1',
+            processor.C1
+        );
+
+        pointOcclusionFS = addConstants(
+            pointOcclusionFS,
+            'C2',
+            processor.C2
+        );
+
+        pointOcclusionFS = addConstants(
+            pointOcclusionFS,
+            'C3',
+            processor.C3
+        );
+
+        pointOcclusionFS = addConstants(
+            pointOcclusionFS,
+            'numSectors',
+            8
+        );
+
+        pointOcclusionFS = addConstants(
+            pointOcclusionFS,
+            'useTriangle',
+            processor.useTriangle
+        );
+
+        pointOcclusionFS = addConstants(
+            pointOcclusionFS,
+            'trianglePeriod',
+            1.0
+        );
+
+        pointOcclusionFS = addConstants(
+            pointOcclusionFS,
+            'dropoutEnabled',
+            processor.dropoutFactor > 1e-6 && context.webgl2);
+
+        return context.createViewportQuadCommand(pointOcclusionFS, {
+            uniformMap : uniformMap,
+            framebuffer : processor._framebuffers.screenSpacePass,
+            renderState : RenderState.fromCache({
+                stencilTest : processor._positiveStencilTest
+            }),
+            pass : Pass.CESIUM_3D_TILE,
+            owner : processor
+        });
+    }
+
+    function densityEdgeCullStage(processor, context) {
+        var uniformMap = {
+            u_pointCloud_ecTexture : function() {
+                return processor._ecTextures[0];
+            },
+            u_neighborhoodVectorSize : function() {
+                return processor.neighborhoodVectorSize;
+            },
+            u_maxAbsRatio : function() {
+                return processor.maxAbsRatio;
+            },
+            u_distanceConstraint : function() {
+                return processor.distanceConstraint;
+            },
+            u_dropoutFactor : function() {
+                return processor.dropoutFactor;
+            }
+        };
+
+        var densityEdgeCullFS = addConstants(
+            DensityEdgeCullPass,
+            'neighborhoodHalfWidth',
+            processor.densityHalfWidth
+        );
+
+        densityEdgeCullFS = addConstants(
+            densityEdgeCullFS,
+            'epsilon8',
+            1e-8
```

Comment:
Thinking about these defines some more, it might be cleaner to keep unchanging defines like this in the shader code, and only call `addConstants` on those that do change.